core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/UnimprovedMoveCountStuckCriterion.java

```diff
@@ -0,0 +1,65 @@
+package ai.timefold.solver.core.impl.localsearch.decider.acceptor.restart;
+
+import java.time.Clock;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.impl.localsearch.decider.reconfiguration.RestartStrategy;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchMoveScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchStepScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+
+/**
+ * Criterion based on the unimproved move count.
+ * It exponentially increases the unimproved move count values that trigger the
+ * {@link RestartStrategy restart} process.
+ */
+public class UnimprovedMoveCountStuckCriterion<Solution_> extends AbstractGeometricStuckCriterion<Solution_> {
+
+    // Multiplier defined through experiments
+    protected static final long UNIMPROVED_MOVE_COUNT_MULTIPLIER = 300_000;
```

Comment:
For ESS or FSR, this will be equivalent to 30-60 seconds.
For smaller problems though, this may be barely a few seconds.
IMO this highlights the need for some sort of a relative criterion.